Sources/GRPCCodeGen/Internal/StructuredSwiftRepresentation.swift

```diff
@@ -503,7 +514,12 @@ enum FunctionKind: Equatable, Codable {
   /// A function or a method. Can be static.
   ///
   /// For example `foo()`, where `name` is `foo`.
-  case function(name: String, isStatic: Bool)
+  case function(
+    name: String,
+    isStatic: Bool,
+    genericType: String? = nil,
+    conformances: [String] = []
```

Comment:
We need these to be able to represent generic functions

Reply:
This isn't quite right: the "conformance" here applies to the generic type, not the function. Also functions can have multiple generic types. 

It looks like the best way to represent this would be by modifying the `FunctionSignatureDescription` to add: a `generics: [ExistingTypeDescription]` property and an optional `var whereClause: WhereClause?`.

That way we can express the `Sendable` requirement as a where clause:

```swift
func foo<R>() -> R where R: Sendable { 
 // ...
}
```